ledfx/virtuals.py

```diff
@@ -990,6 +1020,46 @@ def config(self, new_config):
             VirtualConfigUpdateEvent(self.id, self._config)
         )
 
+        if reactivate_effect:
+            self._reactivate_effect()
+
+    @property
+    def effect_pixel_count(self):
+        """The number of pixels to calculate by effects.
+
+        Can be less than the number of physical pixels (:attr:`~pixel_count`) when
+        pixel grouping (:attr:`~group_size`) is activated.
+        """
+        return self._get_effect_pixel_count(self.pixel_count)
+
+    @property
+    def group_size(self):
+        """The number of physical pixels to group into virtual effect pixels."""
+        grouping = self._config["grouping"]
+
+        if grouping is None or grouping < 1:
+            return 1
+
+        return grouping
+
+    def _get_effect_pixel_count(self, physical_pixel_count):
```

Comment:
Can we use wording of "effective" rather than other, effect is such as loaded term, and seeing effect down in virtuals can be confusing

Reply:
Updated. Wasn't happy with the names, either.

Reply:
Another naming option I considered was "virtual pixel count". Since `virtual.virtual_pixel_count` is redundant, I thought about renaming the virtual count simply to `pixel_count`, and changing the previous `pixel_count` to `physical_pixel_count`. I decided against it due to a lot of changed lines by the rename, but reconsidering it now, it feels quite clear to understand.

Btw., ChatGPT suggests `logical_pixel_count`, `virtual_pixel_count`, `effective_pixel_count`, `transformed_pixel_count`, `composite_pixel_count`, or `mapped_pixel_count`.

Reply:
Hmmmmm I'm not hugely adverse to changing pixel_count to "physical_pixel_count", but given we are actually unlikely to ever need to touch the effective_pixel_count stuff in normal use/development (since it's all handled transparently further up) maybe this is the best way forward.

I'm happy with this for now - I am gonna merge :)